applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -8,406 +8,193 @@
+        auto sorted_ids = rPair.first;
+        std::ranges::sort(sorted_ids);
+        return std::make_pair(sorted_ids, rPair.first);
+    });
+}
 
-    // check 1D elements, note that this has to happen after procedures to find 2 and 3d neighbours are alredy performed, such that 1D elements are only added
+void FindNeighbourElementsOfConditionsProcess::FindNeighbouringElementsForAllBoundaryTypes()
+{
+    auto generate_generic_boundaries = [](const auto& rGeometry) {
+        return rGeometry.GenerateBoundariesEntities();
+    };
+    auto generate_points   = [](const auto& rGeometry) { return rGeometry.GeneratePoints(); };
+    auto generate_edges_3d = [](const auto& rGeometry) {
+        return rGeometry.LocalSpaceDimension() == 3 ? rGeometry.GenerateEdges()
+                                                    : PointerVector<Geometry<Node>>();
+    };
+    auto generate_edges_1d = [](const auto& rGeometry) {
+        return rGeometry.LocalSpaceDimension() == 1 ? rGeometry.GenerateEdges()
+                                                    : PointerVector<Geometry<Node>>();
+    };
+
+    // Note the order in the generators: the 1D elements are only added
     // as neighbours when the condition is not neighbouring 2D or 3D elements
-    this->CheckIf1DElementIsNeighbour(FacesMap);
+    const std::vector<std::function<PointerVector<Geometry<Node>>(const Geometry<Node>&)>> boundary_generators = {
+        generate_generic_boundaries, generate_points, generate_edges_3d, generate_edges_1d};
 
-    // check that all of the conditions belong to at least an element. Throw an error otherwise (this is particularly useful in mpi)
-    all_conditions_visited = true;
-    for (const auto& rCond : mrModelPart.Conditions()) {
-        if (rCond.IsNot(VISITED)) {
-            all_conditions_visited = false;
-            KRATOS_INFO("Condition without any corresponding element, ID ") << rCond.Id() << std::endl;
-        }
+    for (const auto& r_boundary_generator : boundary_generators) {
+        FindConditionNeighboursBasedOnBoundaryType(r_boundary_generator);
+        if (AllConditionsHaveAtLeastOneNeighbour()) return;
     }
-    KRATOS_ERROR_IF_NOT(all_conditions_visited)
-        << "Some conditions found without any corresponding element" << std::endl;
-
-    KRATOS_CATCH("")
 }
 
-bool FindNeighbourElementsOfConditionsProcess::CheckIfAllConditionsAreVisited() const
+void FindNeighbourElementsOfConditionsProcess::FindConditionNeighboursBasedOnBoundaryType(auto generate_boundaries)
 {
-    const auto& r_conditions = mrModelPart.Conditions();
-
-    // Check if all conditions are visited
-    return std::all_of(r_conditions.begin(), r_conditions.end(),
-                       [](const auto& r_cond) { return r_cond.Is(VISITED); });
+    for (auto& r_element : mrModelPart.Elements()) {
+        const auto& rGeometryElement    = r_element.GetGeometry();
+        const auto  rBoundaryGeometries = generate_boundaries(rGeometryElement);
```

Comment:
Just to comply with the Kratos Style Guide (and to make the first name a bit more logical to me):
```suggestion
        const auto& r_element_geometry  = r_element.GetGeometry();
        const auto  boundary_geometries = generate_boundaries(r_element_geometry);
```

Reply:
Good point, done (also the argument of the function)!